### PR TITLE
Addition of ROS2PoseControl gem

### PR DESCRIPTION
**Pull request by Fireronin**
Code moved to seperate gem

### CODE REVIEW

File: Gems/ROS2PoseControl/Code/Include/ROS2PoseControl/ROS2PoseControlBus.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+#pragma once
+
+#include <ROS2PoseControl/ROS2PoseControlTypeIds.h>
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Interface/Interface.h>
+
+namespace ROS2PoseControl
```
Comment (michalpelka) [inline Gems/ROS2PoseControl/Code/Include/ROS2PoseControl/ROS2PoseControlBus.h:9]: - This bus is empty. Please remove it.

File: Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControlSystemComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    {
+    }
+
+    void ROS2PoseControlSystemComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+    }
+
+    ROS2PoseControlSystemComponent::ROS2PoseControlSystemComponent()
+    {
+        if (ROS2PoseControlInterface::Get() == nullptr)
```
Comment (michalpelka) [inline Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControlSystemComponent.cpp:45]: this bus is not used.

File: Gems/ROS2PoseControl/gem.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+    "gem_name": "ROS2PoseControl",
```
Comment (michalpelka) [inline Gems/ROS2PoseControl/gem.json:2]: Please fill those information.

**Review by michalpelka — COMMENTED** [review]
More or less nitpicking

**Review by michalpelka — APPROVED** [review]
My comments are nitpicking.

File: Gems/ROS2PoseControl/Code/Include/ROS2PoseControl/ROS2PoseControlConfiguration.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        AZ::Crc32 isTrackingModeTF2Visibility() const;
+
+        AZ::Crc32 isTrackingModePoseMessagesVisibility() const;
+
+        enum class TrackingMode {
+            PoseMessages,
+            TF2
+        };
+
```
[the remark this answers is not in the document]
Comment (Fireronin) [reply Gems/ROS2PoseControl/Code/Include/ROS2PoseControl/ROS2PoseControlConfiguration.h:24]: Huh, why ? Like I'm actually using those fileds, this looks like a class but it's a struct

File: Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+                                                                        m_topic);
+        m_poseSubscription = ros2Node->create_subscription<geometry_msgs::msg::PoseStamped>(
+            namespaced_topic_name.data(),
+            m_configuration.m_poseTopicConfiguration.GetQoS(),
+            [this](const geometry_msgs::msg::PoseStamped::SharedPtr msg) {
+                if (m_configuration.m_tracking_mode != ROS2PoseControlConfiguration::TrackingMode::PoseMessages || !
+                    m_isTracking) {
+                    return;
+                }
+                const AZ::Transform transform = ROS2::ROS2Conversions::FromROS2Pose(msg->pose);
```
[the remark this answers is not in the document]
Comment (Fireronin) [reply Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp:108]: Ok

File: Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        if (m_configuration.m_tracking_mode == ROS2PoseControlConfiguration::TrackingMode::TF2) {
+            m_poseSubscription.reset();
+            m_tf_buffer = std::make_unique<tf2_ros::Buffer>(ros2Node->get_clock());
+            m_tf_listener = std::make_shared<tf2_ros::TransformListener>(*m_tf_buffer);
+            AZ::TickBus::Handler::BusConnect();
+        } else if (m_configuration.m_tracking_mode == ROS2PoseControlConfiguration::TrackingMode::PoseMessages) {
+            m_tf_buffer.reset();
+            m_tf_listener.reset();
+            AZ::TickBus::Handler::BusDisconnect();
```
[the remark this answers is not in the document]
Comment (Fireronin) [reply Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp:47]: Fair

**Pull request merged**